File: root/tags/2.1.4/source/modules/strategy/src/test/sample_data/inputs/SuggestTrades.java
import java.math.BigDecimal;

import org.marketcetera.trade.Equity;
import org.marketcetera.strategy.java.Strategy;
import org.marketcetera.trade.Factory;
import org.marketcetera.trade.OrderSingle;
import org.marketcetera.trade.OrderType;
import org.marketcetera.trade.Side;
import org.marketcetera.trade.TimeInForce;

/**
 * Test strategy that suggests trades.
 *
 * @author <a href="mailto:colin@marketcetera.com">Colin DuPlantis</a>
 * @version $Id$
 * @since 1.0.0
 */
public class SuggestTrades
    extends Strategy
{
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStart()
     */
    @Override
    public void onStart()
    {
        doSuggestion();
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStop()
     */
    @Override
    public void onStop()
    {
        doSuggestion();
    }
    /**
     * Executes a single suggestion test based on test parameters.
     */
    private void doSuggestion()
    {
        String orderShouldBeNull = getParameter("orderShouldBeNull");
        if(orderShouldBeNull != null) {
            suggestTrade(null,
                         new BigDecimal("100.00"),
                         "suggestion");
        } else {
            OrderSingle suggestedOrder = Factory.getInstance().createOrderSingle();
            suggestedOrder.setAccount(getParameter("account"));
            String orderType = getParameter("orderType");
            if(orderType != null) {
                suggestedOrder.setOrderType(OrderType.valueOf(orderType));
            }
            String price = getParameter("price");
            if(price != null) {
                suggestedOrder.setPrice(new BigDecimal(price));
            }
            String quantity = getParameter("quantity");
            if(quantity != null) {
                suggestedOrder.setQuantity(new BigDecimal(quantity));
            }
            String side = getParameter("side");
            if(side != null) {
                suggestedOrder.setSide(Side.valueOf(side));
            }
            String symbol = getParameter("symbol");
            if(symbol != null) {
                suggestedOrder.setInstrument(new Equity(symbol));
            }
            String timeInForce = getParameter("timeInForce");
            if(timeInForce != null) {
                suggestedOrder.setTimeInForce(TimeInForce.valueOf(timeInForce));
            }
            String scoreString = getParameter("score");
            BigDecimal score;
            if(scoreString != null) {
                score = new BigDecimal(scoreString);
            } else {
                score = null;
            }
            if(getParameter("doOrder") != null) {
                if(getParameter("sendOrderShouldBeNull") != null) {
                    send(null);
                } else {
                    send(suggestedOrder);
                }
            } else {
                suggestTrade(suggestedOrder,
                             score,
                             getParameter("identifier"));
            }
        }
    }
}

File: modules/strategy/src/test/sample_data/inputs/suggest_trades.rb
require 'java'
java_import org.marketcetera.strategy.ruby.Strategy
java_import java.math.BigDecimal
java_import org.marketcetera.trade.Factory
java_import org.marketcetera.trade.OrderType
java_import org.marketcetera.trade.Side
java_import org.marketcetera.trade.TimeInForce
java_import org.marketcetera.trade.Equity

class SuggestTrades < Strategy
    def on_start
        do_suggest
    end
    def on_stop
        do_suggest
    end
    def do_suggest
        orderShouldBeNull = get_parameter("orderShouldBeNull")
        if(orderShouldBeNull != nil)
            suggest_trade(nil, BigDecimal.new("100.00"), "suggestion")
        else
            suggestedOrder = Factory.getInstance().createOrderSingle()
            suggestedOrder.setAccount(get_parameter("account"))
            orderType = get_parameter("orderType")
            if(orderType != nil)
                suggestedOrder.setOrderType(OrderType.valueOf(orderType))
            end
            price = get_parameter("price")
            if(price != nil)
                suggestedOrder.setPrice(BigDecimal.new(price))
            end
            quantity = get_parameter("quantity")
            if(quantity != nil)
                suggestedOrder.setQuantity(BigDecimal.new(quantity))
            end
            side = get_parameter("side")
            if(side != nil)
                suggestedOrder.setSide(Side.valueOf(side))
            end
            symbol = get_parameter("symbol")
            if(symbol != nil)
                suggestedOrder.setInstrument(Equity.new(symbol))
            end
            timeInForce = get_parameter("timeInForce")
            if(timeInForce != nil)
                suggestedOrder.setTimeInForce(TimeInForce.valueOf(timeInForce))
            end
            scoreString = get_parameter("score")
            if(scoreString != nil)
                score = BigDecimal.new(scoreString)
            else
                score = nil
            end
            suggest_trade(suggestedOrder, score, get_parameter("identifier"))
        end
    end
end
